src/test/groovy/unit/org/pih/warehouse/requisition/RequisitionServiceSpec.groovy

```diff
@@ -0,0 +1,159 @@
+        Requisition persistedRequisition = service.saveRequisition(requisition)
+
+        when:
+        service.issueRequisition(persistedRequisition, Mock(User), Mock(Person), comment)
+
+        then:
+        thrown(ValidationException)
+    }
+
+    void 'issueRequisition should change the requisition status to issued'() {
+        given:
+        Requisition requisition = new Requisition(id: 1)
+        String comment = "Comment to issued requisition"
+
+        and:
+        Picklist.metaClass.static.findByRequisition = { Requisition foundRequisition -> return new Picklist(requisition: foundRequisition)
+        }
+        Requisition persistedRequisition = service.saveRequisition(requisition)
+
+        and:
+        service.inventoryService.saveLocalTransfer(_ as Transaction) >> true
+
+        when:
+        service.issueRequisition(persistedRequisition, Mock(User), Mock(Person), comment)
+
+        then:
+        notThrown(ValidationException)
+        1 * inventoryService.generateTransactionNumber()
+        persistedRequisition.status == RequisitionStatus.ISSUED
+    }
+
+    void 'rollbackRequisition should change requisition status to #requisitionStatus when requisition has status #currentStatus'() {
+        given:
+        Requisition requisition = new Requisition(id: 1,
+                status: currentStatus as RequisitionStatus,
+                issuedBy: issuedBy as Person,
+                dateIssued: dateIssued as Date,)
+
+        when:
+        service.rollbackRequisition(requisition)
+
+        then:
+        requisition.status == requisitionStatus
+        requisition.issuedBy == null
+        requisition.dateIssued == null
+
+        where:
+        currentStatus              || requisitionStatus          | issuedBy     | dateIssued
+        RequisitionStatus.CHECKING || RequisitionStatus.CHECKING | null         | null
+        RequisitionStatus.ISSUED   || RequisitionStatus.CHECKING | Mock(Person) | new Date()
+    }
+
+    void 'cloneRequisition should return copy of the passed requisition'() {
+        given:
+        Requisition requisition = new Requisition(id: 1,
+                name: 'Requisition',
+                version: 1,
+                requestedBy: Mock(Person),
+                description: 'Description',
+                dateRequested: new Date(),
+                requestedDeliveryDate: new Date(),
+                lastUpdated: new Date(),
+                status: RequisitionStatus.CHECKING,
+                type: RequisitionType.ADHOC,
+                origin: Mock(Location),)
+
+        when:
+        Requisition copyOfRequisition = service.cloneRequisition(requisition)
+        Map jsonOfOriginalRequisition = requisition.toJson()
+        Map jsonOfCopiedRequisition = copyOfRequisition.toJson()
+
+        then:
+        'Copy of ' + jsonOfOriginalRequisition['name'] == jsonOfCopiedRequisition['name']
+        jsonOfOriginalRequisition.remove('name')
+        jsonOfCopiedRequisition.remove('name')
```

Comment:
kinda weird but I can't think of a better solution off the top of my head outside of just asserting on each key of the map.